### PR TITLE
Login banner too persistent

### PR DESCRIPTION
**Pull request by Lukewh**
Fixes https://github.com/juju/juju-gui/issues/2781

### CODE REVIEW

**Comment by Lukewh** [comment]
## QA
- Load the GUI up incognito/ private browsing
- /login should show the new notification **without the dismiss button**
- Go to /new clicking on the login button in the header should show the new notification **with the dismiss button**
- Add a charm/ bundle to the canvas and click deploy. Clicking on the login button within DF should show the new notification **with the dismiss button**
- After dismissing the login notification, it should reappear if you click login again.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7234/

File: jujugui/static/gui/src/app/components/notification/notification.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('notification', function() {
+
+  juju.components.Notification = React.createClass({
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/notification/notification.js]: Please add a docstring describing the goals and the behavior of this component.

File: jujugui/static/gui/src/app/app.js [hunk 2]
```diff
@@ -402,29 +402,26 @@ YUI.add('juju-gui', function(Y) {
...
-          notification.appendChild(document.createTextNode('.'));
-          document.body.appendChild(notification);
+          const holder = document.getElementById('login-notification');
+          const content = (
+            <span>To proceed with the authentication,
+              please accept the pop up window or&nbsp;
+              <a href={url} target="_blank">click here</a>.</span>);
+          const dismiss = this.state.current.root === 'login' ?
+            null :
+            () => {ReactDOM.unmountComponentAtNode(holder);};
```
Comment (frankban) [inline jujugui/static/gui/src/app/app.js]: Better putting the function body in its own line? [shrug]

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7237/

**Comment by Lukewh** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

File: jujugui/static/gui/src/app/components/notification/notification.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+YUI.add('notification', function() {
+
+  /**
+    Renders a new vanilla style notification
+    (https://docs.vanillaframework.io/en/patterns/notification). Can be
+    dismissed if a 'dismiss' function is passed, otherwise just displays
+    'content' in the desired style.
+  */
+  juju.components.Notification = React.createClass({
+
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/notification/notification.js]: missing `displayName`

File: jujugui/static/gui/src/app/store/env/api.js [hunk 4]
```diff
@@ -633,10 +633,6 @@ YUI.add('juju-env-api', function(Y) {
       @return {undefined} Sends a message to the server only.
     */
     loginWithMacaroon: function(bakery, callback) {
-      if (this.pendingLoginResponse) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/store/env/api.js]: These values are relied on in many places, why was it removed? Have you resolved the pending login race condition here somehow?

Comment (hatched) [inline jujugui/static/gui/src/app/store/env/api.js]: After discussion I now see why this was removed.

**Pull request merged**